spec/features/reservations_spec.rb

```diff
@@ -0,0 +1,330 @@
+      str = reserver ? 'for other user successfully' : 'successfully'
+
+      it str do
+        visit root_path
+        change_reserver(reserver) if reserver
+        add_item_to_cart(@eq_model)
+        update_cart_start_date(Time.zone.today)
+        due_date = Time.zone.today + 1.day
+        update_cart_due_date(due_date)
+        # request catalog since our update cart methods only return the cart
+        # partial from the JS update (so the 'Reserve' link wouldn't be there)
+        visit root_path
+        click_link 'Reserve', href: new_reservation_path
+
+        # set reserver to current user if unset, check confirmation page
+        reserver = reserver ? reserver : @current_user
+        expect(page).to have_content reserver.name
+        expect(page).to have_content Time.zone.today.to_s(:long)
+        expect(page).to have_content due_date.to_s(:long)
+        expect(page).to have_content @eq_model.name
+        expect(page).to have_content 'Create Reservation'
+        click_button 'Finalize Reservation'
+
+        # check that reservation was created with correct dates
+        query = Reservation.for_eq_model(@eq_model).for_reserver(reserver.id)
+        expect(query.count).to eq(1)
+        expect(query.first.start_date).to eq(Time.zone.today)
+        expect(query.first.due_date).to eq(due_date)
+        expect(query.first.approval_status).to eq('auto')
+      end
+    end
+
+    shared_examples 'can create reservation request' do |reserver|
+      let(:reserver) { reserver }
+      str = reserver ? 'for other user successfully' : 'successfully'
+
+      it str do
+        visit root_path
+        change_reserver(reserver) if reserver
+        add_item_to_cart(@eq_model)
+        update_cart_start_date(Time.zone.today)
+        # violate length validation
+        bad_due_date =
+          Time.zone.today + (@eq_model.maximum_checkout_length + 1).days
+        update_cart_due_date(bad_due_date)
+        # request catalog since our update cart methods only return the cart
+        # partial from the JS update (so the 'Reserve' link wouldn't be there)
+        visit root_path
+        click_link 'Reserve', href: new_reservation_path
+
+        # set reserver to current user if unset, check confirmation page
+        reserver = reserver ? reserver : @current_user
+        expect(page).to have_content reserver.name
+        expect(page).to have_content Time.zone.today.to_s(:long)
+        expect(page).to have_content bad_due_date.to_s(:long)
+        expect(page).to have_content @eq_model.name
+        expect(page).to have_content 'File Reservation Request'
+        find(:xpath, "//textarea[@id='reservation_notes']").set 'Because'
+        click_button 'Submit Request'
+
+        # check that reservation was created with correct dates
+        # can't use for_eq_model since it calls finalized
+        query = Reservation.where(equipment_model_id: @eq_model.id)
+                .for_reserver(reserver.id)
```

Comment:
Hmmm, to be honest the `for_eq_model` scope should probably **only** check the equipment model, but we'd have to update all of the current uses of that scope to explicitly call `finalized`. Does that sounds reasonable? I can squeeze that in here, then.